src/ClientData/include/ClientData/CallstackEvent.h

```diff
@@ -27,6 +27,17 @@ class CallstackEvent {
   uint32_t thread_id_;
 };
 
+inline bool operator==(const orbit_client_data::CallstackEvent& lhs,
+                       const orbit_client_data::CallstackEvent& rhs) {
+  return lhs.timestamp_ns() == rhs.timestamp_ns() && lhs.callstack_id() == rhs.callstack_id() &&
+         lhs.thread_id() == rhs.thread_id();
+}
+
+template <typename H>
+H AbslHashValue(H h, const orbit_client_data::CallstackEvent& o) {
+  return H::combine(std::move(h), o.timestamp_ns(), o.callstack_id(), o.thread_id());
+}
```

Comment:
Nit: It's better to declare these as "hidden friends" in the class itself because that means they won't take part in name lookups and will be only be reachable through ADL. They have a couple of subtle advantages. To me the most important one is lower compile times.

Details: https://abseil.io/tips/99 https://abseil.io/tips/152

Reply:
One more: It's good practice to also define `operator!=` when `operator==` and implement it by calling `operator==`.

Reply:
Thanks for the explanation! I finally know now why the examples use `friend`... Done (also `!=`).